cmd/kubelet/app/server.go

```diff
@@ -1173,19 +1174,26 @@ func parseResourceList(m map[string]string) (v1.ResourceList, error) {
 		switch v1.ResourceName(k) {
 		// CPU, memory, local storage, and PID resources are supported.
 		case v1.ResourceCPU, v1.ResourceMemory, v1.ResourceEphemeralStorage, pidlimit.PIDs:
-			if v1.ResourceName(k) != pidlimit.PIDs || utilfeature.DefaultFeatureGate.Enabled(features.SupportNodePidsLimit) {
```

Comment:
Probably a personal preference thing here... but do me it feels a little bit weird to use a switch statement here, given there are only two cases and one of them doesn't have any linked actions.

What about (not valid go):

```
validResourceNames := {v1.ResourceCPU, ...}
if v1.ResourceName(k) not in validResourceNames {
    // do default
}

Reply:
I think using a switch is essentially the only way to do that, but I agree that this feels kinda yanky.

Because of the nature of huge page resources (`hugepages-` prefixed resources), doing so is kinda difficult, unless we explicitly use the `capacity` from the node. We can remove this validation, and do all the validation inside `pkg/kubelet/cm/node_container_manager_linux.go` so that we can use the node capacity, and can make it cleaner.